### DISCUSSION TITLE
Quranic recitation

### DISCUSSION THREAD

**Discussion by maq796113** [openai/whisper#413] [Show and tell]
I thinking about using it to make a real-time live feedback on Quranic recitation using Graphene to Phoneme converters...

**Comment by OmarWKH** [comment]
Have a look at https://www.tarteel.ai/ for inspiration

**Reply by maq796113** [reply]
I have looked into it

**Reply by muhaseenaa** [reply]
there are excellent online Quran learning services for kids, and [MeeM Academia](https://meemacademia.com/) is one of the trusted platforms. It offers personalized and engaging classes specially designed for young learners. With qualified female tutors, fun activities, and a child-friendly approach, it’s a great way for children to learn Quran online at their own pace. Parents looking for a safe and effective learning environment will find [MeeM Academia](https://meemacademia.com/) a reliable choice

**Comment by Ashraf-Ali-aa** [comment]
@maq796113 have a look at https://github.com/tarekeldeeb/DeepSpeech-Quran it uses tarteel.ai data models, I was looking into using the data from tarteel to train whisper to transcribe Quran audio, using https://huggingface.co/blog/fine-tune-whisper

**Reply by maq796113** [reply]
I am using Brain  Speech

**Comment by Ashraf-Ali-aa** [comment]
@maq796113 if you're interested we can work together, I am trying to train whisper on quran audio

**Reply by maq796113** [reply]
Yes I am in fact working on it.

**Reply by Ashraf-Ali-aa** [reply]
@maq796113 Alhamdulillah that's good, if you got a GitHub page or repo I can see if I can help, currently I am creating a dataset on huggingface 

https://huggingface.co/datasets/ashraf-ali/quran-data

**Reply by maq796113** [reply]
Is there way I can contact you?
For now I didn't do much, I am working for senior project. I didn't do much except for make use of Natural Language Toolkit to find the BiLingual Evaluation Understudy score.... 
https://colab.research.google.com/drive/1j2eta-5ySTks0tZvKnMvmZCCEDrAKuPo?authuser=2#scrollTo=eG5vTiJgYsRi


you can go over this colab

**Reply by Ashraf-Ali-aa** [reply]
@maq796113 message me by joining the quran.com discord server  https://discord.gg/FxRWSBfWxn or using my username Ashraf.Ali#0533 , also I can't access the colab URL

**Reply by maq796113** [reply]
I have contacted you on discord

**Reply by abodacs** [reply]
@maq796113 
I can't access the colab URL
 can you fix it?
 
 Thanks

**Reply by BasselAshi** [reply]
I'm very late to the party but I am looking for a model to run for transcribing Quran and I am not sure where to get started. LLM models are a gray area for me.

**Reply by amerrmismarr** [reply]
> Is there way I can contact you? For now I didn't do much, I am working for senior project. I didn't do much except for make use of Natural Language Toolkit to find the BiLingual Evaluation Understudy score.... https://colab.research.google.com/drive/1j2eta-5ySTks0tZvKnMvmZCCEDrAKuPo?authuser=2#scrollTo=eG5vTiJgYsRi
> 
> you can go over this colab

Hello, I am working on the same project, are you still interested in working with someone? Please let me know

**Comment by msis** [comment]
Tarteel put out datasets and a couple of models out.

Check them the org page on HF: https://huggingface.co/tarteel-ai

**Comment by brilliant-ember** [comment]
I found this dataset for Quran recitations by different readers, this version is resampled to 16khz https://www.openslr.org/132/
And this version is the original audio at various sampling rates: https://www.kaggle.com/datasets/bigguyubuntu/quran-ayat-speech-to-text
It's a very good dataset, I had problems with their "tsv" files so I created my own Arabic labels and put it in a csv, it was easy because they have the ayat numbered in the dataset.

**Reply by maq796113** [reply]
how are you planning on using this dataset?

**Comment by Ed-Tivrusky-IV** [comment]
Salam bothers, I'm also considering implement this idea on Unity for a real-time Quran tracking/translation AR application. Let me know if you guys wanna work together!

**Comment by fathimaakk** [comment]
**Meem Quran Classes** is an educational platform dedicated to teaching the Quran, offering online classes for students of all ages. These classes typically include Quran reading, Tajweed (pronunciation rules), memorization (Hifz), and Islamic studies. The focus is on personalized, one-on-one sessions or small group lessons, often led by qualified male and female tutors. The aim is to make Quran learning accessible, flexible, and spiritually enriching for everyone, especially for those living in non-Muslim countries.

**Comment by Walter456951** [comment]
This is a really interesting idea, especially if Graphene-to-phoneme conversion can be used to provide real-time feedback during Quranic recitation. It could potentially help learners identify pronunciation issues and improve their recitation more effectively while practicing. Alongside the technology side, consistent guidance from experienced teachers can also make a big difference. I recently came across [quran classes](https://learnquranonlineusa.com/) that offer structured Quran learning and recitation guidance, and they seem worth exploring for anyone who wants additional support. Combining useful technology with proper instruction could make Quran learning more interactive, accessible, and beneficial for learners at different levels.
